src/System.Management.Automation/FormatAndOutput/common/PSStyle.cs

```diff
@@ -42,6 +42,24 @@ public enum ProgressView
     /// </summary>
     public sealed class PSStyle
     {
+        /// <summary>
+        /// Colorizes message by prepending given color escape, then resets if requested.
+        /// </summary>
+        /// <param name="message">
+        /// String to colorize.
+        /// </param>
+        /// <param name="colorEscape">
+        /// ANSI color and/or font escape sequence.
+        /// </param>
+        /// <param name="reset">
+        /// If given, the color is reset to default (not necessarily previous) setting.
+        /// </param>
+        /// <returns>
+        /// Colorized string.
+        /// </returns>
+        public static string Colorize(string message, string colorEscape, bool reset = true)
```

Comment:
I think it may be better to call this `Decorate` rather than `Colorize` since you can use ANSI for non-color decorations like underline, bold, italics, etc...